lib/binance/futures.ex

```diff
@@ -0,0 +1,1755 @@
+    arguments =
+      %{}
+      |> with_optional_arg("symbol", symbol)
+
+    case HTTPClient.get_binance_unsigned(
+           "/fapi/v1/indexInfo",
+           arguments,
+           api_key: Application.get_env(:binance, :futures_api_key),
+           secret_key: Application.get_env(:binance, :futures_secret_key),
+           base_url: Application.get_env(:binance, :futures_end_point)
+         ) do
+      {:ok, data} ->
+        {:ok, Enum.map(data, &Binance.Futures.CompositeIndexSymbolInfo.new/1)}
+
+      {:error, error} ->
+        {:error, error}
+    end
+  end
+
+  @doc """
+  Multi-Assets Mode Asset Index
+
+  ## Examples
+
+      BinanceFutures.get_multi_assets_mode_asset_index()
+  """
+  def get_multi_assets_mode_asset_index(symbol \\ nil) do
+    arguments =
+      %{}
+      |> with_optional_arg("symbol", symbol)
+
+    case HTTPClient.get_binance_unsigned(
+           "/fapi/v1/assetIndex",
+           arguments,
+           api_key: Application.get_env(:binance, :futures_api_key),
+           secret_key: Application.get_env(:binance, :futures_secret_key),
+           base_url: Application.get_env(:binance, :futures_end_point)
+         ) do
+      {:ok, data} when is_list(data) ->
+        {:ok, Enum.map(data, &Binance.Futures.MultiAssetsModeAssetIndex.new/1)}
+
+      {:ok, data} ->
+        {:ok, Binance.Futures.MultiAssetsModeAssetIndex.new(data)}
+
+      {:error, error} ->
+        {:error, error}
+    end
+  end
+
+  @doc """
+  Account information
+
+  ## Examples
+
+      BinanceFutures.get_account()
+  """
+  def get_account(receive_window \\ nil, timestamp \\ nil) do
+    arguments =
+      %{}
+      |> with_optional_args([{"recvWindow", receive_window}, {"timestamp", timestamp}])
+
+    case HTTPClient.get_binance(
+           "/fapi/v2/account",
+           arguments,
+           api_key: Application.get_env(:binance, :futures_api_key),
+           secret_key: Application.get_env(:binance, :futures_secret_key),
+           base_url: Application.get_env(:binance, :futures_end_point)
+         ) do
+      {:ok, data} -> Binance.Futures.Account.new(data)
+      {:error, error} -> {:error, error}
+    end
+  end
+
+  @doc """
+  Get Income History
+
+  ## Examples
+
+      BinanceFutures.get_income_history()
+  """
+  def get_income_history(
+        symbol \\ nil,
+        income_type \\ nil,
+        limit \\ 100,
+        start_time \\ nil,
+        end_time \\ nil,
+        receive_window \\ nil,
+        timestamp \\ nil
+      ) do
+    arguments =
+      %{}
+      |> with_optional_args([
+        {"symbol", symbol},
+        {"incomeType", income_type},
+        {"limit", limit},
+        {"startTime", start_time},
+        {"endTime", end_time},
+        {"recvWindow", receive_window},
+        {"timestamp", timestamp}
+      ])
+
+    case HTTPClient.get_binance(
+           "/fapi/v1/income",
+           arguments,
+           api_key: Application.get_env(:binance, :futures_api_key),
```

Comment:
será que não conseguimos fazer isso uma vez só? 

Quando temos algo assim no elixir: `x: 10, y: 20` é uma [Keyword list](https://elixir-lang.org/getting-started/keywords-and-maps.html) que podemos juntar com o operador `++` sem perder nada. Como esse passo de ir pra configs vai ser repetitivo, é uma boa simplificar com uma call montando uma Keyword List e depois só juntar ela com os outros argumentos quando relevante.

uma nota, se usássemos as macros do Tesla, isso se tornaria um plug, bem mais fácil de testar, mockar etc.

Reply:
Como o código foi gerado automaticamente, nem me dei muita conta sobre a repetição infinita 🤦

Extrai essas opções pra um module attribute. Não sei se é a melhor solução, mas já tá bem melhor